src/wallet/models.py

```diff
@@ -46,8 +49,31 @@ def by_direction(self, direction):
         else:
             return self.incoming()
 
+    def _latest_stamps(self):
+        # Filter out the latest status timestamps for every transaction
+        stamps = self.annotate(latest_stamp=models.Max('states__date_created'))
+        return stamps.values_list('latest_stamp', flat=True)
+
+    def by_status(self, status=None):
```

Comment:
It could be written more concise:
```python
def by_status(self, status=None):
    qs = self.filter(states__date_created__in=self._latest_stamps())
    if status is not None:
        qs = qs.filter(states__status=status)
    return qs
```